src/cocotb_tools/runner.py

```diff
@@ -784,10 +784,10 @@ class Icarus(Runner):
 
     .. admonition:: Simulator-specific Usage
 
-       * ``hdl_toplevel`` argument to :meth:`.build` is *required*.
-       * ``waves=True`` *must* be given to :meth:`.build` if either ``waves`` or ``gui`` are to be used during :meth:`.test`.
-       * ``timescale`` argument to :meth:`.build` must be given to support dumping the command file.
-       * Does not support the ``pre_cmd`` argument to :meth:`.test`.
+       * ``hdl_toplevel`` argument to :meth:`Runner.build` is *required*.
+       * ``waves=True`` *must* be given to :meth:`Runner.build` if either ``waves`` or ``gui`` are to be used during :meth:`Runner.test`.
+       * ``timescale`` argument to :meth:`Runner.build` must be given to support dumping the command file.
+       * Does not support the ``pre_cmd`` argument to :meth:`Runner.test`.
```

Comment:
Why is this necessary? IIRC when using `.` prefix for links, Sphinx does a contextual lookup, so `test` and `build` should link to the current class. Using `Runner` without an anchor only works if you are setting `currentmodule` which we generally don't want to do.

Reply:
When I added `cocotb_tools.pytest.hdl.HDL` that is also defining own `.build()` and `.test()` methods, sphinx started spamming me with these warnings from each defined runner class:

```plaintext
/home/tblazejczyk/workspace/github/tymonx/cocotb/.nox/docs/lib/python3.12/site-packages/cocotb_tools/runner.py:docstring of cocotb_tools.runner.Icarus:8: WARNING: more than one target found for cross-reference 'test': cocotb_tools.pytest.hdl.HDL.test, cocotb_tools.runner.Runner.test [ref.python]
```

This was with :meth:`.build` and :meth:`.test` in the `runner.rst` file.

How I should proceed to fix that correctly :thinking:

Reply:
@ktbarrett I have completely reverted `src/cocotb_tools/runner.py` to original state:

- c5b1b107ddd20d38ed1af3d4ebab101a0d5758e4
- a279335c71113234a1c518b01feb334438f19fc8

You can also confirm that on Pull Request files changed tab.